lib/3scale/backend/job_fetcher.rb

```diff
@@ -85,6 +85,8 @@ def try_pop_from_queue(max=nil)
       def wait_pop_from_queue
         queue, encoded = @redis.blpop(*@queues, timeout: @fetch_timeout)
         [queue, [encoded]] if encoded
+      rescue RedisClient::ReadTimeoutError => _e
+        # Do nothing, we expect frequent timeouts from the sync client when the queue is empty
```

Comment:
IIRC, when `@fetch_timeout` is reached, the method will return `nil`, not raise. A read timeout on the other hand is a low-level TCP timeout that shouldn't happen regularly.

On the other hand, we don't need to be very much bothered by a connection timeout, if we can just reconnect. It's good to be resilient in case of network glitches.

I would suggest though to log the error, can be without a trace, just dome text like:
> Que redis (ip of server if possible to get) connection timeout at #{__FILE__}:#{__LINE__}

In this way we can have an indication of network stability and see whether particular servers or locations are more problematic. This will help debugging and improving the cluster setup.

Reply:
Replied at https://github.com/3scale/apisonator/pull/384#issuecomment-2109715939